packages/react-debug-tools/src/ReactDebugHooks.js

```diff
@@ -149,8 +149,19 @@ function nextHook(): null | Hook {
 }
 
 function readContext<T>(context: ReactContext<T>): T {
+  if (context === GetPrimitiveStackCacheContext) {
+    // This is a read for filling the primitive stack cache.
+    // There's no sensible value to return.
+    return (null: any);
+  }
+  if (currentContextDependency === null) {
+    return context._currentValue;
```

Comment:
Ideally we'd throw an error here since that would be a bug if the context dependencies don't line up with the context reads. However, doing so would break APIs like `ReactDebugTools.inspectHooks` where we don't have access to the current fiber (we have tests that would break if we throw here). Which also means that `ReactDebugTools.inspectHooks` would not have support for `useFormStatus`.